R/plot_sampling_hist.R

```diff
@@ -17,5 +17,53 @@
 #' samples <- draw_samples(pop, 3, c(1, 10))
 #' plot_sampling_hist(pop, samples, var_name, c(1, 10), 3)
 plot_sampling_hist <- function(pop, samples, var_name, n_s, reps){
+  if (!"data.frame" %in% c(class(samples))) stop("Samples should be a data frame or a tibble")
+  if (!is.numeric(reps) == TRUE) stop("Number of replications should be a numerical value, a vector with length 1")
+  for (i in n_s){
+    if (class(i) != "numeric") stop("Samples' sizes should be a list or a vector with only numeric values")
+  }
+
+  samples <-
+    samples %>%
+    ungroup()
+
+  x_min <-
+    samples %>%
+    select({{var_name}}) %>%
+    unlist() %>%
+    quantile(0.05)
+
+  x_max <-
+    samples %>%
+    ungroup() %>%
+    select({{var_name}}) %>%
+    unlist() %>%
+    quantile(0.95)
+
+
+  summary <-
+    samples %>%
+    group_by(replicate, size, rep_size) %>%
+    summarise(mean = mean({{var_name}}))
+
+
+  sampling_dist<- list()
+  for (i in 1:length(n_s)){
+    sampling_dist[[ i ]] <-
+      summary %>%
+      filter(size == n_s[i]) %>%
+      ggplot() + geom_histogram(aes(mean, ..density..)) +
+      ggtitle(paste("sample size", n_s[i])) +
+      coord_cartesian(xlim = c(x_min, x_max)) +
+      theme(plot.title = element_text(size = 10))
+    if (i > 1){
+      sampling_dist[[i]] <- sampling_dist[[i]] + theme(axis.title.y=element_blank(),
+                                                       axis.text.y=element_blank(),
+                                                       axis.ticks.y=element_blank())
+    }
+
+  }
+
+  return(grid.arrange(grobs = sampling_dist, nrow=1, top = "Sampling Distribution Histograms"))
```

Comment:
Note:  it is difficult to do tests on grid.arrange / grob objects. I ended up splitting my code into two functions: one that creates the histograms (so you can test labels, axis mapping, etc), and one that arranges them in a grid

Reply:
Not required, just a suggestion

Reply:
updated it! Thanks!